test/Handler/StreamHandlerTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * Horde Log package
+ *
+ * This package is based on Zend_Log from the Zend Framework
+ * (http://framework.zend.com).  Both that package and this
+ * one were written by Mike Naberezny and Chuck Hagenbuch.
+ *
+ * @author     Rafael te Boekhorst <boekhorstb1@b1-systems.de>
+ * @category   Horde
+ * @license    http://www.horde.org/licenses/bsd BSD
+ * @package    Log
+ * @subpackage UnitTests
+ */
+
+namespace Horde\Log\Test\Handler;
+
+use Horde\Log\Handler\StreamHandler;
+
+use Horde_Cli;
+use Horde\Log\Formatter\CliFormatter;
+use Horde\Log\Formatter\SimpleFormatter;
+
+use PHPUnit\Framework\TestCase;
+
+use Horde\Log\LogException;
+use Horde_Log;
+
+use Horde\Log\LogMessage;
+use Horde\Log\LogLevel;
+
+class StreamHandlerTest extends TestCase
+{
+    public function setUp(): void
+    {
+        date_default_timezone_set('America/New_York');
+        $formater = new SimpleFormatter('%timestamp% %levelName%: %message%' . PHP_EOL);
+
+        $this->level1 = new LogLevel(Horde_Log::ALERT, 'Alert');
+        $this->message1 = 'this is an emergency!';
+        $this->logMessage1 = new LogMessage($this->level1, $this->message1, ['timestamp' => date('c')]);
+        $this->formattedMessage1 = $this->logMessage1->formatMessage([$formater]);
+    }
+
+    public function testConstructorThrowsWhenResourceIsNotStream()
+    {
+        $this->expectException(LogException::class);
+        $resource = xml_parser_create();
+        $test = new StreamHandler($resource);
+        xml_parser_free($resource); # closes the resource, has no other effects (this is only for pre-php8)
+    }
+
+    // public function testConstructorWithValidStream()
+    // {
+    //     $stream = fopen('php://memory', 'a');
+    //     new StreamHandler($stream);
+    //     $this->markTestSkipped('No Exception expected.');
+    // }
+
+    // public function testConstructorWithValidUrl()
+    // {
+    //     new StreamHandler('php://memory');
+    //     $this->markTestSkipped('No Exception expected.');
+    // }
```

Comment:
If I understand correctly, these only exist to check that the constructor does not throw any errors with these parameters.
Instread of skipping them, you could add an anotation:
```
    /**
     * @doesNotPerformAssertions
     */
    public function testConstructorWithValidUrl()
    {
        new StreamHandler('php://memory');
    }
```

Reply:
ok I will add something like that :)

Reply:
I would prefer you checking for getting a LogHandler instance rather than using that annotation.

Reply:
> 

A ok, will  do that tomorrow then!